frontend/src/views/LoginView.tsx

```diff
@@ -1,23 +1,45 @@
-// import UserForm from "../components/UserForm";
-// import useAuth from "../hooks/Auth";
+import UserForm from "../components/UserForm";
+import useAuth from "../hooks/Auth";
 import { useNavigate } from "react-router-dom";
 
 export default function LoginView() {
-    // const auth = useAuth();
     const navigate = useNavigate();
+    const auth = useAuth(); //Access contect using useAuth
+
+    //Signin function to handle user sign in
+    const onSubmit = async (userId: string, password: string): Promise<boolean> =>{
+        try {
+            // API call to authenticate the user
+            const response = await fetch(`/api/v1/auth/sign-in?userid=${encodeURIComponent(userId)}&password=${encodeURIComponent(password)}`, {
+                method: "POST",
+            });
+
+            //Check if signup was successful
+            if(response.ok){
+                const data = await response.json(); // Assuming the token is returned in JSON format
+                const token = { data: data.token }; // Wrap token to match Token interface
```

Comment:
The token should:tm: be returned directly, there is no wrapper JSON object. I think that the auth.login() function expects a string, so you might be better off gettting the body of the response directly as a string and passing that in instead of decoding the json